Spigot-API-Patches/0258-Add-recipe-to-cook-events.patch

```diff
@@ -0,0 +1,60 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Thonk <30448663+ExcessiveAmountsOfZombies@users.noreply.github.com>
+Date: Wed, 6 Jan 2021 12:05:29 -0800
+Subject: [PATCH] Add recipe to cook events
+
+
+diff --git a/src/main/java/org/bukkit/event/block/BlockCookEvent.java b/src/main/java/org/bukkit/event/block/BlockCookEvent.java
+index be7af5440bf9923f0c9c84efa4d70a89337a2f96..25274440c33e1c1352113cfe3dae9824de892423 100644
+--- a/src/main/java/org/bukkit/event/block/BlockCookEvent.java
++++ b/src/main/java/org/bukkit/event/block/BlockCookEvent.java
+@@ -14,12 +14,31 @@ public class BlockCookEvent extends BlockEvent implements Cancellable {
+     private final ItemStack source;
+     private ItemStack result;
+     private boolean cancelled;
++    private final org.bukkit.inventory.CookingRecipe recipe; // Paper
+ 
++    @Deprecated // Paper
+     public BlockCookEvent(@NotNull final Block block, @NotNull final ItemStack source, @NotNull final ItemStack result) {
++        // Paper start
++        this(block, source, result, null);
++    }
++
++    /**
++     * Gets the cooking recipe associated with this event.
++     *
++     * @return the recipe
++     */
++    @org.jetbrains.annotations.Nullable
++    public org.bukkit.inventory.CookingRecipe getRecipe() {
++        return recipe;
```

Comment:
idk if I like having the getter inbetween the ctors, but I see why you did it, so 🤷 
what do others think?

Reply:
Sacrifice some diff for prettiness instead :)